diff --git a/modules/luci-lua-runtime/Makefile b/modules/luci-lua-runtime/Makefile
index 8ad2887f0..a43d570bc 100644
--- a/modules/luci-lua-runtime/Makefile
+++ b/modules/luci-lua-runtime/Makefile
@@ -8,7 +8,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=luci-lua-runtime
 PKG_SRC_VERSION:=26.231.50176~e9cdc0a
-PKG_GITBRANCH:=LuCI c2000max-v36.01-head branch
+PKG_GITBRANCH:=LuCI c2000max-v36.10-head branch
 
 LUCI_TYPE:=mod
 LUCI_BASENAME:=lua-runtime
