--- a/mt_wifi/common/cmm_rdm_mt.c
+++ b/mt_wifi/common/cmm_rdm_mt.c
@@ -7337,7 +7337,6 @@ VOID WrapDfsRddReportHandle(/* handle th
 
 #ifdef CONFIG_MAP_SUPPORT
 #ifdef MAP_R2
-		UCHAR band;
 		UCHAR first_wdev = TRUE;
 		UCHAR Channel_for_radar = 0;
 		UCHAR bw_for_radar = 0;
@@ -7529,7 +7528,6 @@ VOID WrapDfsRddReportHandle(/* handle th
 #if defined(OFFCHANNEL_SCAN_FEATURE) && defined(MAP_R2)
 	if (IS_MAP_ENABLE(pAd)) {
 		Rsp.Action = DFS_RADAR_HIT;
-		band = (pAd->CommonCfg.dbdc_mode) ? DBDC_BAND1 : DBDC_BAND0;
 		memcpy(Rsp.ifrn_name, pAd->ScanCtrl.if_name, IFNAMSIZ);
 		Rsp.data.operating_ch_info.cfg_ht_bw = wlan_config_get_ht_bw(&pAd->ApCfg.MBSSID[MAIN_MBSSID].wdev);
 		Rsp.data.operating_ch_info.cfg_vht_bw = wlan_config_get_vht_bw(&pAd->ApCfg.MBSSID[MAIN_MBSSID].wdev);
--- a/mt_wifi/os/linux/cfg80211/cfg80211_rx.c
+++ b/mt_wifi/os/linux/cfg80211/cfg80211_rx.c
@@ -1199,7 +1199,6 @@ BOOLEAN CFG80211_HandleMgmtFrame(RTMP_AD
 
 	if (pWdev) {
 		pNetDev = pWdev->if_dev;
-		OpMode = pWdev->wdev_type == WDEV_TYPE_AP ? OPMODE_AP : OPMODE_STA;
 		if ((pWdev->wdev_type == WDEV_TYPE_AP)
 			&& ap_rx_mgmt_drop(pAd, (FRAME_CONTROL *)pRxBlk->FC))
 			return TRUE; /* drop packet */
