--- a/files-ucode/usr/share/ucode/wifi/iface.uc
+++ b/files-ucode/usr/share/ucode/wifi/iface.uc
@@ -23,8 +23,12 @@ export function parse_encryption(config,
 
 	config.auth_type = encryption[0] ?? 'none';
 
+	let is_eht = wildcard(dev_config?.htmode, 'EHT*');
 	let wpa3_pairwise = config.wpa_pairwise;
-	if (wildcard(dev_config?.htmode, 'EHT*') || wildcard(dev_config?.htmode, 'HE*'))
+	if (is_eht)
+		config.ieee80211be = true;
+
+	if (is_eht || wildcard(dev_config?.htmode, 'HE*'))
 		wpa3_pairwise = 'GCMP-256 ' + wpa3_pairwise;
 
 	switch(config.auth_type) {
@@ -170,12 +174,18 @@ export function wpa_key_mgmt(config) {
 		append_value(config, 'wpa_key_mgmt', 'SAE');
 		if (config.ieee80211r)
 			append_value(config, 'wpa_key_mgmt', 'FT-SAE');
+		if (config.ieee80211be && config.rsn_override) {
+			config.rsn_override_key_mgmt = config.wpa_key_mgmt;
+			append_value(config, 'rsn_override_key_mgmt_2', 'SAE-EXT-KEY');
+			if (config.ieee80211r)
+				append_value(config, 'rsn_override_key_mgmt_2', 'FT-SAE-EXT-KEY');
+		}
 		break;
 
 	case 'psk-sae':
 		append_value(config, 'wpa_key_mgmt', 'SAE');
 		if (config.ieee80211r)
 			append_value(config, 'wpa_key_mgmt', 'FT-SAE');
 
 		if (config.rsn_override) {
 			config.rsn_override_key_mgmt = config.wpa_key_mgmt;
